webgoat-lessons/sql-injection/src/main/java/org/owasp/webgoat/sql_injection/introduction/SqlInjectionLesson9.java

```diff
@@ -58,12 +59,14 @@ public AttackResult completed(@RequestParam String name, @RequestParam String au
 
     protected AttackResult injectableQueryIntegrity(String name, String auth_tan) {
         StringBuffer output = new StringBuffer();
-        String query = "SELECT * FROM employees WHERE last_name = '" + name + "' AND auth_tan = '" + auth_tan + "'";
+        String query = "SELECT * FROM employees WHERE last_name = '" + name + "' AND auth_tan = " + "?";
         try (Connection connection = dataSource.getConnection()) {
             try {
-                Statement statement = connection.createStatement(TYPE_SCROLL_SENSITIVE, CONCUR_UPDATABLE);
+                PreparedStatement statement = connection.prepareStatement(query, TYPE_SCROLL_SENSITIVE,
+                        CONCUR_UPDATABLE);
                 SqlInjectionLesson8.log(connection, query);
-                ResultSet results = statement.executeQuery(query);
+                statement.setString(1, auth_tan);
+                ResultSet results = statement.executeQuery();
```

Comment:
# Mend Code Security Check

### New finding (2 of 4)
> The Mend Code Security Check of your branch failed because of a XSS finding in this line.

<table role='table'><thead><tr><th>Severity</th><th>Vulnerability Type</th><th>CWE</th><th>File</th><th>Data Flows</th><th>Date</th></tr></thead><tbody><tr><td><a href='#'><img src='https://whitesource-resources.whitesourcesoftware.com/high_vul.png?' width=19 height=20></a> High</td><td>XSS</td><td>

[CWE-79](https://cwe.mitre.org/data/definitions/79.html)

</td><td>

[Servers.java:71](https://github.com/NormanWenzelWSS/WebGoat/blob/34571d56f2035edee961f17ac05cd67975db0cbc/webgoat-lessons/sql-injection/src/main/java/org/owasp/webgoat/sql_injection/mitigation/Servers.java#L71)

</td><td>1</td><td>2023-07-04 09:25am</td></tr><tr><td colspan='6'><details><summary><a href='#'><img src='https://saas.mend.io/sast/favicon.png' width=15 height=15></a> Vulnerable Code</summary>

https://github.com/NormanWenzelWSS/WebGoat/blob/34571d56f2035edee961f17ac05cd67975db0cbc/webgoat-lessons/sql-injection/src/main/java/org/owasp/webgoat/sql_injection/mitigation/Servers.java#L66-L71

<details>
<summary>1 Data Flow/s detected</summary></br>


<details>
<summary>View Data Flow 1</summary>

https://github.com/NormanWenzelWSS/WebGoat/blob/34571d56f2035edee961f17ac05cd67975db0cbc/webgoat-lessons/sql-injection/src/main/java/org/owasp/webgoat/sql_injection/mitigation/Servers.java#L67

https://github.com/NormanWenzelWSS/WebGoat/blob/34571d56f2035edee961f17ac05cd67975db0cbc/webgoat-lessons/sql-injection/src/main/java/org/owasp/webgoat/sql_injection/mitigation/Servers.java#L71

https://github.com/NormanWenzelWSS/WebGoat/blob/34571d56f2035edee961f17ac05cd67975db0cbc/webgoat-lessons/sql-injection/src/main/java/org/owasp/webgoat/sql_injection/mitigation/Servers.java#L71

</details>
</details>
</details>

<details>

<summary><a href='#'><img src='https://integration-api.securecodewarrior.com/explorer/favicon-32x32.png' width=15 height=15></a> Secure Code Warrior Training Material</summary>

<ul><li>Training<ul><li>

[Secure Code Warrior SQL Injection Training](https://portal.securecodewarrior.com/?utm_source=partner-integration:mend&partner_id=mend#/contextual-microlearning/web/injection/sql/java/vanilla)</li>

</ul></li>

<li>Videos<ul><li>

[Secure Code Warrior SQL Injection Video](https://media.securecodewarrior.com/v2/module_01_sql_injection.mp4)</li>

</ul></li>

<li>Further Reading<ul><li>

[OWASP SQL Injection Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html)</li>

<li>

[OWASP SQL Injection](https://owasp.org/www-community/attacks/SQL_Injection)</li>

<li>

[OWASP Query Parameterization Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Query_Parameterization_Cheat_Sheet.html)</li>

</ul></li>

</ul>

</details>

<details>

<summary><a href='#'><img src='https://whitesource-resources.whitesourcesoftware.com/cvss3.png' width=15 height=15></a> Suppress Finding</summary>

<!-- SAST-SUPPRESS-START -->
- [ ] ... as False Alarm
- [ ] ... as Acceptable Risk
<!-- SAST-SUPPRESS-END -->

</details>

</td></tr>
</tbody></table>